src/game_inventory.cpp

```diff
@@ -1,21 +1,53 @@
 #include "game.h"
 
+#include "cata_utility.h"
 #include "inventory_ui.h"
 #include "player.h"
 #include "item.h"
 #include "itype.h"
 
+item_location inv_internal( player &p, const inventory_selector_preset &preset,
+                            const std::string &title, int radius,
+                            const std::string &none_message )
+{
+    p.inv.restack( &p );
+    p.inv.sort();
+
+    inventory_pick_selector inv_s( p, preset );
+
+    inv_s.set_title( title );
```

Comment:
The selectors themselves. The callers populate it by means of its public interface. The `inventory_selector` is a "slave" class and I prefer it to remain so.